executor/brie.go

```diff
@@ -307,10 +502,36 @@ func (e *BRIEExec) Next(ctx context.Context, req *chunk.Chunk) error {
 
 	bq := globalBRIEQueue
 
+	taskExecuted := false
+
 	e.info.connID = e.ctx.GetSessionVars().ConnectionID
 	e.info.queueTime = types.CurrentTime(mysql.TypeDatetime)
-	taskCtx, taskID := bq.registerTask(ctx, e.info)
-	defer bq.cancelTask(taskID)
+
+	// create new session so processInfo will not be overwritten
+	s, err := CreateSessionForBRIEFunc(e.ctx)
```

Comment:
Use `e.ctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL()` can also protect the ProcessInfo from been overwritten.